### PR TITLE
Small bug fix for InfoMax

### PR DESCRIPTION
**Pull request by neworderofjamie**
1. ``size_t`` is unsigned meaning that ``bestIndex`` logic was broken
2. ``InfoMaxRotater`` didn't implement ``getImageDifferences``

### CODE REVIEW

File: navigation/infomax.h [hunk 1]
```diff
@@ -203,27 +203,56 @@ class InfoMaxRotater : public InfoMax<FloatType>
...
     //------------------------------------------------------------------------
+    template<class... Ts>
+    const std::vector<FloatType> &getImageDifferences(Ts &&... args) const
+    {
+        calcImageDifferences(std::forward<Ts>(args)...);
+        return m_RotatedDifferences;
+    }
+
     template<class... Ts>
     auto getHeading(Ts &&... args) const
```
Comment (alexdewar) [inline navigation/infomax.h]: Does this need a ``&`` to make sure we get a reference and not a copy?

Comment (neworderofjamie) [reply navigation/infomax.h]: Where exactly? possible places to put ``&`` is almost as fun a C++ quiz question as possible places to put ``const``...

Comment (alexdewar) [reply navigation/infomax.h]: Oh yeah, sorry :-p

I *think* the ``auto`` in the return type might be resolved to a value type cf. a reference.

I meant like this:
``auto &getHeading(Ts &&... args) const``

It also looks like you can use ``decltype(auto)`` for this:
https://en.cppreference.com/w/cpp/language/auto

Comment (neworderofjamie) [reply navigation/infomax.h]: Ahh ok - I don't **think** ``getHeading`` should return a reference though - it's returning a tuple which, amongst other things, include a reference to ``m_RotatedDifferences``.

Comment (alexdewar) [reply navigation/infomax.h]: Ah, my bad. I guess in that case you def won't want to return by reference.

**Pull request merged**